app.py

```diff
@@ -51,7 +59,39 @@ def post(self):
         }
         return json, 201
 
+#user login POST
+class LoginResource(Resource):
+    def post(self):
+        user = User.query.filter_by(email=request.json['email']).first()
+
```

Comment:
We want both the email and the username to be validated here - you'll want to add another query and slightly change the variable names, as well as add an `or` statement to the if/else logic
something like:
```python
user_by_email = User.query.filter_by(email=request.json['email']).first()
user_by_username = User.query.filter_by(username=request.json['username']).first()

if user_by_email == None or user_by_username == None:
  return  { error message }
```